packages/@aws-cdk/cx-api/lib/asset-manifest-artifact.ts

```diff
@@ -0,0 +1,24 @@
+import * as cxschema from '@aws-cdk/cloud-assembly-schema';
+import * as path from 'path';
+import { CloudArtifact } from './cloud-artifact';
+import { CloudAssembly } from './cloud-assembly';
+
+/**
+ * Asset manifest is a description of a set of assets which need to be built and published
+ */
+export class AssetManifestArtifact extends CloudArtifact {
```

Comment:
Do we really need these in `cx-api`? I am concerned about adding more stuff to this module, which I think we should completely deprecate given it's extremely sensitive posture between the CLI and the framework. I rather we remove stuff from it then add more stuff if possible.

@iliapolo what do you think?

Reply:
@eladb Well, since this class is eventually used to instantiate a `CloudAssembly` object, it makes sense that its here. 

What does strikes me as odd is this: 

```typescript
 if (!properties.file) {
      throw new Error('Invalid AssetManifestArtifact. Missing "file" property');
 }
```

This should be part of the regular schema validation mechanism. And actually the `file` property of this artifact should somehow be defined in the schema. This happens because the `properties` property of an `ArtifactManifest` is defined as:

https://github.com/aws/aws-cdk/blob/ac3b330c71ef258afd145b86fd90a06db5d1c990/packages/%40aws-cdk/cloud-assembly-schema/lib/schema.ts#L306

Even though in reality we expect it to have a specific structure. Same can be said for `CloudFormationStackArtifact`.

Reply:
Yeah I just mirrored what was there already.